File: be/PodoCareWebsite/src/main/java/com/podocare/PodoCareWebsite/model/order/Order.java
package com.podocare.PodoCareWebsite.model.order;

import com.podocare.PodoCareWebsite.model.Brand_Supplier.Supplier;
import com.podocare.PodoCareWebsite.model.VatRate;
import com.podocare.PodoCareWebsite.model.product.product_category.product_instances.EquipmentProductInstance;
import com.podocare.PodoCareWebsite.model.product.product_category.product_instances.SaleProductInstance;
import com.podocare.PodoCareWebsite.model.product.product_category.product_instances.ToolProductInstance;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.ToString;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "internal_order")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne
    @JoinColumn(name = "supplier_id", nullable = false)
    private Supplier supplier;


    private Long orderNumber;
    private Date orderDate;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL)
    @JsonManagedReference
    @ToString.Exclude
    private List<OrderProduct> orderProducts = new ArrayList<>();

    private VatRate shippingVatRate;
    private Double shippingCost = 0.0;
    private Double totalNet = 0.0;
    private Double totalVat = 0.0;
    private Double totalValue = 0.0;
}

File: fe/podocare_pro/src/models/order.ts
import { OrderProduct, NewOrderProduct } from "./order-product";
import { Supplier } from "./supplier";
import { VatRate } from "./vatrate";

export interface Order {
    id: number;
    supplier: Supplier;
    orderNumber: number;
    orderDate: string;
    orderProducts: OrderProduct[];
    shippingVatRate: VatRate;
    shippingCost: number;
    totalNet: number;
    totalVat: number;
    totalValue: number;
}

export interface NewOrder {
    supplier?: Supplier;
    orderDate?: string;
    orderProducts?: NewOrderProduct[];
    shippingVatRate: VatRate;
    shippingCost: number;
    totalNet: number;
    totalVat: number;
    totalValue: number;
}

export interface OrderFilterDTO {
    supplierIds?: number[] | null;
    dateFrom?: string | null;
    dateTo?: string | null;
}
